src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php

```diff
@@ -78,7 +78,11 @@ protected function createController($controller)
      */
     protected function instantiateController($class)
     {
-        $controller = parent::instantiateController($class);
+        if ($this->container->has($class)) {
+            $controller = $this->container->get($class);
+        } else {
+            $controller = parent::instantiateController($class);
+        }
```

Comment:
This is not possible without a larger refactoring because `createController` is never called in this case: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php#L70

An alternative is to refactor the whole `getController` method but it's risky and intrusive. It's why I've chosen this simple fix.